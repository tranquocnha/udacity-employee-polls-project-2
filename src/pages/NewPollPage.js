import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { useState } from "react";
import { handleAddQuestion } from "../store/actions/questions";

const NewPollPage = ({ dispatch }) => {
  const navigate = useNavigate();
  const [firstOption, setFirstOption] = useState("");
  const [secondOption, setSecondOption] = useState("");

  const handleFirstOptionChange = (e) => {
    setFirstOption(e.target.value);
  };

  const handleSecondOptionChange = (e) => {
    setSecondOption(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(handleAddQuestion(firstOption, secondOption));
    navigate("/");
  };

  return (
    <div className="container-fluid">
      <h1 className="font-bold ">New Poll</h1>
      <form onSubmit={handleSubmit}>
        <div className="form-outline mb-4">
          <label htmlFor="firstOption"
                           data-testid="firstOptionLabel" 
          className="form-label">First Option</label>
          <input
            id="firstOption"
            data-testid="firstOption"
            value={firstOption}
            onChange={handleFirstOptionChange}
            type="text"
            className="form-control"
          />
        </div>

        <div className="form-outline mb-4">
          <label className="form-label"
          htmlFor="secondOption"
          data-testid="secondOptionLabel">Second Option</label>
          <input
            type="text"
            id="secondOption"
            data-testid="secondOption"
            value={secondOption}
            onChange={handleSecondOptionChange}
            className="form-control"
          />
        </div>

        <button
          className="btn btn-primary btn-block mb-4"
          type="submit"
          data-testid="submit"
        >
          Submit
        </button>
      </form>
    </div>
  );
};

export default connect()(NewPollPage);
