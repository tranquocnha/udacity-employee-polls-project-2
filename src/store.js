import {configureStore} from '@reduxjs/toolkit';
import authedUser from "./store/reducers/authedUser";
import users from "./store/reducers/users";
import questions from "./store/reducers/questions";

export const store = configureStore({
  reducer: {
    authedUser,
    users,
    questions,
  },
});
