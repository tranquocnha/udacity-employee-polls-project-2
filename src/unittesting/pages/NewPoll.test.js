import {fireEvent, render} from "@testing-library/react";
import { store } from "../../store";
import { Provider } from "react-redux";
import { BrowserRouter} from "react-router-dom";
import React from "react";
import NewPollPage from "../../pages/NewPollPage";

describe("NewPolPage", () => {
    it("should render the NewPolPage component", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewPollPage/>
                </BrowserRouter>
            </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });

    it("should display all elements in Poll page", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewPollPage/>
                </BrowserRouter>
            </Provider>
        );

        const firstOptionLabelElement = component.getByTestId("firstOptionLabel");
        const firstOptionInputElement = component.getByTestId("firstOption");
        const secondOptionLabelElement = component.getByTestId("secondOptionLabel");
        const secondOptionInputElement = component.getByTestId("secondOption");
        const submitButtonElement = component.getByTestId("submit");

        expect(firstOptionLabelElement.textContent).toBe("First Option");
        expect(secondOptionLabelElement.textContent).toBe("Second Option");
        expect(submitButtonElement.textContent).toBe("Submit");

        fireEvent.change(firstOptionInputElement, {target: {value: 'First'}});
        fireEvent.change(secondOptionInputElement, {target: {value: 'Second'}});
        expect(firstOptionInputElement.value).toBe("First");
        expect(secondOptionInputElement.value).toBe("Second");
    });
});
