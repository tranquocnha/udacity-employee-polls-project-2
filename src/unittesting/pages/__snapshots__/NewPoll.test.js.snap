// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPolPage should render the NewPolPage component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid"
      >
        <h1
          class="font-bold "
        >
          New Poll
        </h1>
        <form>
          <div
            class="form-outline mb-4"
          >
            <label
              class="form-label"
              data-testid="firstOptionLabel"
              for="firstOption"
            >
              First Option
            </label>
            <input
              class="form-control"
              data-testid="firstOption"
              id="firstOption"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-outline mb-4"
          >
            <label
              class="form-label"
              data-testid="secondOptionLabel"
              for="secondOption"
            >
              Second Option
            </label>
            <input
              class="form-control"
              data-testid="secondOption"
              id="secondOption"
              type="text"
              value=""
            />
          </div>
          <button
            class="btn btn-primary btn-block mb-4"
            data-testid="submit"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid"
    >
      <h1
        class="font-bold "
      >
        New Poll
      </h1>
      <form>
        <div
          class="form-outline mb-4"
        >
          <label
            class="form-label"
            data-testid="firstOptionLabel"
            for="firstOption"
          >
            First Option
          </label>
          <input
            class="form-control"
            data-testid="firstOption"
            id="firstOption"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-outline mb-4"
        >
          <label
            class="form-label"
            data-testid="secondOptionLabel"
            for="secondOption"
          >
            Second Option
          </label>
          <input
            class="form-control"
            data-testid="secondOption"
            id="secondOption"
            type="text"
            value=""
          />
        </div>
        <button
          class="btn btn-primary btn-block mb-4"
          data-testid="submit"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
